stripe-crypto-onramp/src/main/java/com/stripe/android/crypto/onramp/OnrampInteractor.kt

```diff
@@ -444,10 +451,56 @@ internal class OnrampInteractor @Inject constructor(
             else -> OnrampCheckoutResult.Failed(PaymentFailedException())
         }
     }
+
+    /**
+     * Gets the platform publishable key from state, or fetches it if not available.
+     * Returns null if fetch fails or key is null.
+     */
+    private suspend fun getOrFetchPlatformKey(): String? {
+        val currentConsumerSecret = consumerSessionClientSecret()
+        val cachedKey = _state.value.platformKeyCache
+
+        // Check if we have a valid cached key for the current consumer session
+        if (cachedKey != null && cachedKey.consumerSessionClientSecret == currentConsumerSecret) {
+            return cachedKey.publishableKey
+        }
+
+        // Fetch platform settings if not available or consumer session changed
+        val platformSettingsResult = cryptoApiRepository.getPlatformSettings(
+            consumerSessionClientSecret = currentConsumerSecret,
+            countryHint = null
+        )
+        if (platformSettingsResult.isFailure) {
+            return null
```

Comment:
We'll want to surface the error, especially since this will error if the `cscs` is invalid (missing KYC info).